oneShow.js

```diff
@@ -0,0 +1,246 @@
+        episodeSummary.toLowerCase().includes(searchString.value.toLowerCase())
+      ) {
+        allShows[i].classList.remove("hide-it");
+        showingEpisodeCount++;
+      } else {
+        allShows[i].classList.add("hide-it");
+      }
+    }
+    showCounter.textContent = `Showing ${showingEpisodeCount} episode(s) out of ${episodeList.length}`;
+  } else {
+    showCounter.textContent = `Showing ${episodeList.length} episode(s) out of ${episodeList.length}`;
+
+    for (let i = 0; i < allShows.length; i++) {
+      allShows[i].classList.remove("hide-it");
+    }
+  }
+}
+
+// jump to episode
+function jumpToEpisode() {
+  let selectList = document.getElementById("selection-list");
+  let position = selectList.value;
+  let selectedId = "episode-card" + position;
+  document.getElementById(selectedId).scrollIntoView();
+}
+
+// build selection list dropdown
+
+function buildEpisodeDropdown(showElem, episodeList) {
+  let episodeSelectionList = document.createElement("select");
+  episodeSelectionList.classList.add("episodes-dropdown");
+  episodeSelectionList.id = "selection-list";
+  showElem.appendChild(episodeSelectionList);
+
+  for (let i = 0; i < episodeList.length; i++) {
+    let optionSeasonPadded = episodeList[i].season.toString().padStart(2, "0");
+    let optionEpisodeNumberPadded = episodeList[i].number
+      .toString()
+      .padStart(2, "0");
+    let optionCodeValue =
+      "S" + optionSeasonPadded + "E" + optionEpisodeNumberPadded + " - ";
+    let optionName = episodeList[i].name;
+
+    let episodeOptionContent = optionCodeValue + optionName;
+
+    let episodeOptionElement = document.createElement("option");
+    episodeOptionElement.textContent = episodeOptionContent;
+    episodeOptionElement.value = i;
+    episodeSelectionList.appendChild(episodeOptionElement);
+  }
+
+  let listListener = document.getElementById("selection-list");
+  listListener.addEventListener("change", jumpToEpisode);
+}
+
+// build search input box
+
+function buildSearchInput(showElem, episodeList) {
+  let searchInput = document.createElement("input");
+  searchInput.classList.add("search-input");
+  searchInput.id = "search-field";
+  showElem.appendChild(searchInput);
+  searchInput.addEventListener("input", function () {
+    searchFilter(episodeList);
+  });
+}
+
+// build search counter
+
+function buildSearchCounter(showElem, episodeList) {
+  let searchCounterText = document.createElement("span");
+  searchCounterText.id = "search-counter";
+  showElem.appendChild(searchCounterText);
+  searchCounterText.textContent = `Showing ${episodeList.length} episode(s) out of ${episodeList.length}`;
+}
+
+// build all episodes
+
+function buildAllEpisodes(showElem, episodeList) {
+  let episodesContainer = document.createElement("div");
+  episodesContainer.classList.add("episodes-container");
+  showElem.appendChild(episodesContainer);
+
+  for (const [i, episode] of episodeList.entries()) {
```

Comment:
All the parts of the code are written in smaller parts, I should start doing this in my code as well. Its easier to read.